### ISSUE TITLE
CSharpPrinter drops parentheses around compound range endpoints, emitting uncompilable a..b

### ISSUE THREAD

**Issue by richlander** [richlander/dotnet-inspect#1424]
## Goal

Over-raise correctness bug for the #1356 burndown (printer-spelling class, like the `checked` and float-conv siblings). The C# printer emits a `RangeExpression`'s endpoints with bare `Expression()` instead of the precedence-wrapping `Operand()`, so a compound range bound (`a + b`, `a * b`, …) is left unparenthesized. The C# range operator `..` binds tighter than additive/multiplicative operators on its operand, so the printed text **re-parses with the wrong grouping and does not compile** — yet fidelity is reported `Full`.

## The over-raise claim being narrowed

The printer assumes a range endpoint can be spelled with bare `Expression()`. That is only true for atoms. `..` is higher-binding than `+`/`-`/`*`/`<<`… on its operands, so `(a + b)..` must keep its parentheses. Every other operand position in the printer routes children through `Operand()` (`CSharpPrinter.cs:1582`), which parenthesizes any non-atom; `RangeExpression` is the one infix/operand site that bypasses it. The very next line, `IndexFromEnd` (`CSharpPrinter.cs:1433`), already uses `Operand()` correctly.

## Minimal counterexample (standard csc, not synthetic)

```csharp
public static int[] SliceStart(int[] arr, int a, int b) => arr[(a + b)..];
public static int[] SliceEnd(int[] arr, int a, int b)   => arr[..(a + b)];
```

Driving the real pipeline (`IrImporter.Import` → `IrPasses.Run` → `CSharpPrinter.Print`) produces, at `Fidelity=Full`:

| source | decompiler emits | re-parses as |
| --- | --- | --- |
| `arr[(a + b)..]` | `arr[a + b..]` | `arr[a + (b..)]` |
| `arr[..(a + b)]` | `arr[..a + b]` | `arr[(..a) + b]` |

Recompiling the decompiled output fails:

```
error CS0019: Operator '+' cannot be applied to operands of type 'int' and 'Range'
error CS0019: Operator '+' cannot be applied to operands of type 'Range' and 'int'
```

So this is invalid C# (validity boss) reported as `Full`. (Also reproduces with `a * b` and other binary endpoints; ternary/coalesce endpoints don't reach this site because csc spills them to a temp before the slice, and atomic bounds — locals, constants, `a.Length` — are already fine.)

## Missing gate

`RangeExpression` endpoints are not routed through the printer's precedence-wrapping `Operand()`.

## First fix shape

In `CSharpPrinter.cs:1432`, emit both endpoints via `Operand()` instead of `Expression()`:

```csharp
RangeExpression r => $"{(r.HasStart ? Operand(r.Start!) : "")}..{(r.HasEnd ? Operand(r.End!) : "")}",
```

`Operand()` wraps non-atoms and leaves atoms (and nested `RangeExpression`/`IndexFromEnd`) bare, so `arr[(a + b)..]` round-trips and `arr[a..]` / `arr[^1..]` are unaffected. This matches the existing `IndexFromEnd` treatment exactly.

## Done signal

- Adversarial fixtures: `arr[(a + b)..]`, `arr[..(a + b)]`, `arr[(a * b)..(c + d)]` recompile (parens preserved).
- Positive canaries unchanged: `arr[a..]`, `arr[..b]`, `arr[a..b]`, `arr[^1..]`, `arr[a.Length..]` still print without redundant parens.

## Notes

Confirmed end-to-end through the product pipeline and round-tripped through csc (CS0019). Single-site printer fix. Printer-spelling correctness class, same family as #1403 (float conv drop) and the `checked`/`unchecked` scoping bug — valid-looking `Full` output that is actually wrong.

**Comment by richlander** [comment]
Claimed from #1356 row 32 on branch fix/issue-1424-range-endpoint-parens. Starting with focused range endpoint precedence fixtures and the narrow printer fix.

**Comment by richlander** [comment]
Opened PR #1495. The fix routes range endpoints through `Operand()`, preserving atomic endpoints while parenthesizing compound bounds like `(i + j)..` and `..(i + j)`; targeted compile-back coverage for the new fixtures is exact.
